swift/swift.py

```diff
@@ -52,102 +61,131 @@ def __init__(self, setup_env: dict, parent=None):
+            info: A dictionary containing app info. Each element is like below:
+              path: A path desired to be added for importing app.
+              module: A name of app module.
+              class: A name of app class.
+              show: Whether its frames is shown at the beginning.
+                True if you want to show the frames, otherwise False.
+              pos: A string that indicates the position of the frames.
+                It should be one of "left", "right", "top", or "bottom"; and is case-sensitive.
+                Otherwise, it will be regarded as default (AllDockWidgetAreas).
+              bus: A list of buses which app subscribes to.
+              args: Additional arguments for custom.
+                If there is no additional arguments, set None.
         """
-        pos_to_area = {
-            "left": Qt.LeftDockWidgetArea,
-            "right": Qt.RightDockWidgetArea,
-            "top": Qt.TopDockWidgetArea,
-            "bottom": Qt.BottomDockWidgetArea
-        }
-        for name, info in setup_app.items():
-            # show frames if the 'show' option is true
-            if info["show"]:
-                for frame in self._apps[name].frames():
-                    dockWidget = QDockWidget(name, self.mainWindow)
-                    dockWidget.setWidget(frame)
-                    area = pos_to_area.get(info["pos"], Qt.AllDockWidgetAreas)
-                    self.mainWindow.addDockWidget(area, dockWidget)
-        self.mainWindow.show()
+        # import the app module
+        with _add_to_path(os.path.dirname(info.get("path", ""))):
+            module = importlib.import_module(info["module"])
+        # create an app
+        cls = getattr(module, info["class"])
+        if "args" in info:
+            app = cls(name, self, **info["args"])
+        else:
+            app = cls(name, self)
+        # set a slot of broadcast signal to router
+        app.broadcastRequested.connect(self._routeToBus)
+        # add the app to the list of subscribers on each bus
+        for bus_name in info["bus"]:
+            self._subscribers[bus_name].add(app)
+        # show frames if the "show" option is true
+        if info["show"]:
+            for frame in app.frames():
+                dockWidget = QDockWidget(name, self.mainWindow)
+                dockWidget.setWidget(frame)
+                area = {
+                    "left": Qt.LeftDockWidgetArea,
+                    "right": Qt.RightDockWidgetArea,
+                    "top": Qt.TopDockWidgetArea,
+                    "bottom": Qt.BottomDockWidgetArea
+                }.get(info["pos"], Qt.AllDockWidgetAreas)
+                self.mainWindow.addDockWidget(area, dockWidget)
+        # store the app
+        self._apps[name] = app
+
+    def destroyBus(self, name: str):
+        """Destroys a global bus.
+        
+        Args:
+            name: A name of the bus to destroy.
+        """
+        self._buses[name].deleteLater()

```

Comment:
It should be deleted from Python (`self._buses.remove(name)` or something, so it can be garbage-collected).
Moreover, the bus should be stopped. Therefore, `self._buses.pop(name).stop()` would be perfect.
Finally, the `_subscribers` should be updated as well (remove the key `name`).

Reply:
Ok, I will! But we should decide how to deal with `_subscribers`.